src/core/search/autocompleteresponsetransformer.js

```diff
@@ -0,0 +1,37 @@
+import AutoCompleteData, { AutoCompleteResult } from '../models/autocompletedata';
+
+/**
+ * A data transformer that takes the response object from an
+ * AutoComplete request and transforms it into a front-end oriented
+ * data structure that our component library and core storage understand.
+ */
+export default class AutoCompleteResponseTransformer {
+  static transformAutoCompleteResponse (response) {
+    const sections = [{
+      results: response.results.map(result => new AutoCompleteResult(result)),
+      resultsCount: response.results.length
+    }];
+    return new AutoCompleteData({
+      sections: sections,
+      queryId: response.queryId,
+      inputIntents: response.inputIntents
+    });
+  }
+
+  static transformFilterAutoCompleteResponse (response) {
```

Comment:
Out of curiosity are we able to reference the types exported from the core in jsdoc? maybe with command + click working in vscode? https://github.com/yext/answers-core/pull/24

Reply:
Yes, it looks like we are able to. Are you saying we should try to use those types here in the code?

Reply:
yes we can reference them in jsdoc! imo it's just a nice quality of life thing

Reply:
something like `@param {AutoCompleteResponse}` or maybe `@param {import('@yext/answers-core').AutoCompleteResponse}`

Reply:
Ah gotcha! Sorry for some reason I was thinking you meant referencing them in the JS code (like `(response: AutoCompleteResponse)`) in the method signature, and I got confused. I'll add some JSdoc and push them up!